src/sdk/object_sdk.js

```diff
@@ -315,7 +315,11 @@ class ObjectSDK {
         const source_ns = await this._get_bucket_namespace(bucket);
         const source_md = await source_ns.read_object_md(source_params, this);
         if (params.tagging_copy) await this._populate_source_object_tagging({ source_params, source_ns, source_md });
-
+        if (ranges && ranges[0].end > source_md.size) {
+            let err = new Error('Invalid Argument');
+            err.code = 'InvalidArgument';
+            throw err;
+        }
```

Comment:
Let's take the normalize_http_ranges and put it outside the if, and add the new check into that func (under a specific flag)